### ISSUE TITLE
Read only populated Date field fail to submit

### ISSUE THREAD

**Issue by brunobuzzi** [orbeon/orbeon-forms#4966]
1) Create a form that call a external service to populate fields (it must include a date field)
2) Put the date field to be populated as read only
3) Try to submit the form

Step 3) will fail, if the field is changed to readOnly false --> the submit works as expected

I will try to upload a gif with the example

Orbeon Forms 2020.1.1.202102012142 PE

<property as="xs:string"  name="oxf.xforms.format.input.date"  value="[D]/[M]/[Y]"/>
<property as="xs:string"  name="oxf.fr.resource.orbeon.*.*.print.formats.date" value="[D]/[M]/[Y]"/>

**Comment by brunobuzzi** [comment]
The first gif "Field Bug 1.gif" is show the bug, the form is not submited.

The second gif the form is edited and the date field is set readOnly false, after this the form is submited correctly. I think this bug is a regression because this was working ok with previous Orbeon versions.

![Field Bug 1](https://user-images.githubusercontent.com/4719282/130369467-bcbe599b-6f3e-4901-9a5c-07810f36772f.gif)
![Field Bug 2](https://user-images.githubusercontent.com/4719282/130369712-0d3b4f8c-47f9-49cf-8320-576a1c35590c.gif)

**Comment by avernet** [comment]
Hi @brunobuzzi, could you try to enable the XForms Inspector and see what the value of the date field is after the action ran? Is it set to a date in the ISO format (e.g. `2021-08-23`)?

**Comment by brunobuzzi** [comment]
The value in Xforms inspector when readOnly=false is [citizenBirth]1960-06-22[/citizenBirth]
The value in Xforms inspector when readOnly=true is [citizenBirth]1960-6-22[/citizenBirth]

The difference is 06 and 6. Maybe this is causing the problem ?

**Comment by avernet** [comment]
Yes, that could explain it. What does the service return? Does it by any chance return `1960-6-22`? Maybe it works when the field is read-write because somehow that value makes a round trip to the "UI", which fixes the missing the `0`, as if the user had entered that value?
